_inc/client/plans/single-product-backup.jsx

```diff
@@ -51,6 +58,80 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
 	);
 }
 
+class SingleProductBackupBody extends React.Component {
+	static propTypes = {
+		backupPlanPrices: PropTypes.object,
+		currencySymbol: PropTypes.string,
+		setSelectedBackupType: PropTypes.func,
+		selectedBackupType: PropTypes.string,
+		upgradeLinks: PropTypes.object,
+	};
+
+	handleSelectedBackupTypeChange = event => {
+		this.props.setSelectedBackupType( event.target.value );
+	};
+
+	render() {
+		const { currencySymbol, backupPlanPrices, selectedBackupType, upgradeLinks } = this.props;
+
+		const upgradeTitles = {
+			'real-time': __( 'Upgrade to Real-Time Backups' ),
+			daily: __( 'Upgrade to Daily Backups' ),
+		};
+
+		const backupOptions = [
+			{
+				type: 'daily',
+				name: __( 'Daily Backups' ),
+				price: backupPlanPrices.jetpack_backup_daily.yearly,
+			},
+			{
+				type: 'real-time',
+				name: __( 'Real-Time Backups' ),
+				price: backupPlanPrices.jetpack_backup_realtime.yearly,
+			},
+		];
+
+		return (
+			<React.Fragment>
+				<p>
+					{ __(
+						'Always-on backups ensure you never lose your site. Choose from real-time or daily backups. {{a}}Which one do I need?{{/a}}',
+						{
+							components: {
+								a: <ExternalLink href="https://jetpack.com/upgrade/backup/" icon iconSize={ 12 } />,
+							},
+						}
+					) }
+				</p>
+				<h4 className="single-product-backup__options-header">{ __( 'Backup options:' ) }</h4>
+				<div className="single-product-backup__radio-buttons-container">
+					{ backupOptions.map( option => (
+						<PlanRadioButton
+							key={ option.type }
+							checked={ option.type === selectedBackupType }
+							currencySymbol={ currencySymbol }
+							planName={ option.name }
+							planPrice={ option.price }
+							onChange={ this.handleSelectedBackupTypeChange }
+							radioValue={ option.type }
+						/>
+					) ) }
+				</div>
+				{ upgradeLinks &&
```

Comment:
Thanks for adding those checks.

I think we still need to check if the `upgradeTitles` object exists first, just like we check for `upgradeLinks`.

What do you think about extracting this whole piece to a separate method like `renderUpgradeLink`. Inside of it, we wouldn't have to build such a long inline condition like we have here. Instead we could just have regular `if` statements that would `return null` if we have no links or titles. Feel free to refactor if you find it useful, it's not a blocker though.

Reply:
upgradeTitles is always going to be defined because it's set to the following value at the top of the function:

```javascript
const upgradeTitles = {
	'real-time': __( 'Upgrade to Real-Time Backups' ),
	daily: __( 'Upgrade to Daily Backups' ),
};
```